### PR TITLE
migrating to hooks: ChangePasswordForm, SocialLoginGroup, ZipRecruite…

### PR DESCRIPTION
**Pull request by chynh**
# Description of changes

Converted ZipRecruiterJobs, join, SocialGroupLogin, change_password, ChangePasswordForm to functional component using hooks

# Issue Resolved

Partially tackled #845

### CODE REVIEW

**Comment by kylemh** [comment]
#### 📝Changed routes:


| [**/join**](https://operation-code-git-fork-chynh-hooks.operation-code.now.sh/join) | [**/profile/change_password**](https://operation-code-git-fork-chynh-hooks.operation-code.now.sh/profile/change_password) |
|:-:|:-:|
|<a href="https://operation-code-git-fork-chynh-hooks.operation-code.now.sh/join">Screenshot of /join</a><br /><sup><a href="https://microlink-cdn.s3.amazonaws.com/s/ruzJJJysb_K6QW/Ze0tbG_628iP8_">(view full size)</a> |<a href="https://operation-code-git-fork-chynh-hooks.operation-code.now.sh/profile/change_password">Screenshot of /profile/change_password</a><br /><sup><a href="https://microlink-cdn.s3.amazonaws.com/s/koeIpslmn_rf4c/xvSp_8Gc7_B6c_">(view full size)</a> |







Commit 4eac03f607f299bc3a1b69d376ab6b362cc7fe7b (https://operation-code-git-fork-chynh-hooks.operation-code.now.sh).

File: components/ChangePasswordForm/ChangePasswordForm.js [hunk 1]
```diff
@@ -56,58 +39,71 @@ export default class ChangePasswordForm extends React.Component {
...
-              <Field
-                type="password"
-                name="newPassword2"
-                label="Confirm Password*"
-                component={Input}
-                disabled={isSubmitting}
-                autoComplete="new-password"
-              />
-            </div>
+ChangePasswordForm.propTypes = {
```
Comment (kylemh) [inline components/ChangePasswordForm/ChangePasswordForm.js]: Can you move these above the function definition?

File: components/SocialLoginGroup/SocialLoginGroup.js [hunk 2]
```diff
@@ -1,58 +1,53 @@
...
-    );
-  }
-}
+
+      {children({ onSuccess, onGoogleFailure })}
+    </div>
+  );
+};
+
+SocialLoginGroup.propTypes = {
```
Comment (kylemh) [inline components/SocialLoginGroup/SocialLoginGroup.js]: Can you move these above the function definition?

File: pages/join.js [hunk 3]
```diff
@@ -9,44 +9,41 @@ import RegistrationForm from 'components/RegistrationForm/RegistrationForm';
...
 
-class Join extends React.Component {
-  static propTypes = {
-    router: object.isRequired,
-  };
-
-  componentDidMount() {
-    const { router } = this.props;
-
+const Join = props => {
```
Comment (kylemh) [inline pages/join.js]: Can you destructure props here?

File: pages/join.js [hunk 4]
```diff
@@ -9,44 +9,41 @@ import RegistrationForm from 'components/RegistrationForm/RegistrationForm';
...
+            </Link>
+            .
+          </p>,
+        ]}
+      />
+    </>
+  );
+};
+
+Join.propTypes = {
```
Comment (kylemh) [inline pages/join.js]: Can you move these above the function definition?

File: pages/profile/change_password.js [hunk 5]
```diff
@@ -7,42 +7,34 @@ import Content from 'components/Content/Content';
 import ChangePasswordForm from 'components/ChangePasswordForm/ChangePasswordForm';
 import styles from '../styles/password_reset.css';
 
-class ChangePassword extends React.Component {
-  state = {
-    didChange: false,
+const ChangePassword = () => {
+  const [didChange, setDidChange] = React.useState(false);
+  const onSuccess = () => {
```
Comment (kylemh) [inline pages/profile/change_password.js]: Very nitpicky, but can you add some newlines between blocks?

Comment (chynh) [reply pages/profile/change_password.js]: done

File: pages/join.js (hunk 3 above)
Comment (chynh) [reply pages/join.js]: done

File: components/ChangePasswordForm/ChangePasswordForm.js (hunk 1 above)
Comment (chynh) [reply components/ChangePasswordForm/ChangePasswordForm.js]: Below is the error I get when I move the prop types above function definition. Do you have any suggestions on a workaround this error???
``
ReferenceError: Cannot access 'SocialLoginGroup' before initialization
``

Comment (kylemh) [reply components/ChangePasswordForm/ChangePasswordForm.js]: You'll just need to convert the arrow functions to vanilla functions

File: pages/join.js (hunk 3 above)
Comment (kylemh) [reply pages/join.js]: I meant: `function Join({ router }) {` then you can delete line 14

File: pages/profile/change_password.js (hunk 5 above)
Comment (kylemh) [reply pages/profile/change_password.js]: My guess is the commit will come in later? I don't see any changes.

Comment (chynh) [reply pages/profile/change_password.js]: Yep. commit will come soon. Just making changes that you requested.

**Pull request merged**